tempesta_fw/server.c

```diff
@@ -59,9 +63,15 @@ static atomic64_t act_sg_n = ATOMIC64_INIT(0);
  * When a server connection is scheduled for connect it increments server's
  * reference count and decrements it after inteded disconnect.
  */
-static LIST_HEAD(sg_list);
-static LIST_HEAD(sg_list_reconfig);
-static DEFINE_RWLOCK(sg_lock);
+#define TFW_SG_HBITS	10
```

Comment:
Maybe this parameter should be configurable (for cases when server groups count is much larger or much less than 1K).

Reply:
We spend 4 pages for both of the hash tables, so reducing hash size won't give us big memory win. There is no need to move the hash size to configuration, since we're able to estimate the optimal hash size by the number of configured server groups. I have no idea whether it has sense to optimize the data structure further to handle 100K and more server groups, since profiling results has shown timers as the main bottleneck.